Port MASTG-TEST-0030 (PendingIntent) to v2 format (#3562)
OWASP/mastg: demos/android/MASVS-PLATFORM/MASTG-DEMO-0x01/MASTG-DEMO-0x01.md

4. **Secure PendingIntent (PASS)**: Creates a `PendingIntent` with `FLAG_IMMUTABLE` and an explicit intent that specifies both the target class and package. This is the recommended secure approach.

{{ MastgTest.kt # MastgTest_reversed.java }}

## Steps

Run the @MASTG-TOOL-0110 rule against the reversed Java code to identify all PendingIntent creation calls.

{{ ../../../../rules/mastg-android-pendingintent-mutable.yml }}

{{ run.sh }}

## Observation

The rule identifies **4 findings** where `PendingIntent` creation APIs are used. This includes both insecure and secure implementations.

{{ output.txt }}

## Evaluation

Review each of the reported instances:

- **Lines 26-31**: FAIL - Uses `PendingIntent.getActivity()` with an implicit intent (`ACTION_VIEW`) and `FLAG_UPDATE_CURRENT` without `FLAG_IMMUTABLE`. A malicious app could intercept this PendingIntent and modify its target.

- **Lines 36-41**: FAIL - Uses `PendingIntent.getService()` with `FLAG_MUTABLE` explicitly set. Even though the intent is explicit, the mutable flag allows modification of unfilled fields.

> - **Lines 46-51**: FAIL - Uses `PendingIntent.getBroadcast()` with an implicit intent (custom action) and no flags. On API < 31, this is mutable by default.

Reviewer: ```suggestion
- **Lines 46-51**: FAIL - Uses `PendingIntent.getBroadcast()` with an implicit intent (custom action) and no flags. On API < 31, the absence of flags results in implicit mutability.
```

Reviewer: - This explains much better that prior to API level 31, passing no flags (`0`) results in mutability because the system simply left the intent mutable by omission, but is not specifically "mutable by default" _(which could be misinterpreted as "FLAG_MUTABLE")_.
